middleware/base-auth.ts

```diff
@@ -0,0 +1,27 @@
+import { Middleware } from '@nuxt/types';
+import { UserType } from '~/types/entities';
+
+const auth: Middleware = (context) => {
+	const user = context.$auth.user;
+	const professorRoutes = ['/professor'];
+	const studentRoutes = ['/student'];
+
+	if (!user || !context.$auth.loggedIn) {
+		context.redirect('/login');
+	} else {
+		if (context.route.path === '/login') {
+			context.redirect('/');
+		}
+
+		if (
```

Comment:
tem que colocar tbm o caminho pro admin não? pq se o cara é adminho ele n vai cair em nenhuma dessas condições

Reply:
O admin não terá restrições, apenas a do primeiro if.

Reply:
O admin não terá restrições, apenas a do primeiro if.

Reply:
sim, mas não tem nenhum contex.redirect fora dos ifs, se o cara for admin o middleware n vai fazer nada (eu acho)